symforce/codegen/backends/python/templates/geo_package/custom_methods/pose2.py.jinja

```diff
@@ -2,16 +2,19 @@
  # SymForce - Copyright 2022, Skydio, Inc.
  # This source code is under the Apache 2.0 license found in the LICENSE file.
  # ---------------------------------------------------------------------------- #}
+{%- import "../../util/util.jinja" as util with context -%}
 
     {# Handwritten methods for Pose2 #}
     {# These will get included into the autogenerated class header. #}
     def __init__(self, R=None, t=None):
-        # type: (Rot2, T.Sequence[float]) -> None
+        # type: (Rot2, T.Union[T.Sequence[float], numpy.ndarray]) -> None
         rotation = R if R is not None else Rot2()
-        position = t if t is not None else [0., 0.]
-        assert isinstance(rotation, Rot2)
+        t = t if t is not None else [0., 0.]
+        {{ util.flatten_if_ndarray("t", 2) | indent(width=8) }}
+        if not isinstance(rotation, Rot2):
+            raise ValueError(f"arg R has type {type(R)}; type {Rot2} expected")
```

Comment:
Thanks. Though, I can't help but notice this passed CI, despite having this in the test:
``` python
        # Test constructor raises a ValueError if a non Rot is passed for R
        with self.assertRaises(ValueError):
            geo_class(R=4)  # type: ignore[arg-type]
```
Makes sense since our tests are run in python3, but it might be worth identifying which tests we want passing in python2 and running those with python2 as well. Maybe there are other things like this I've accidentally committed in the past.

Reply:
100% agree we should be testing our generated code in py2 as well